test/session/SessionSpec.coffee

```diff
@@ -772,3 +772,45 @@ describe 'Session', ->
             resolve()))
 
     .then((() -> done()), (err) -> done(err))
+
+  it 'should limit the number of sessions', (done) ->
+    [alice_ident, bob_ident] = [0, 1].map(-> Proteus.keys.IdentityKeyPair.new())
+    bob_store = new TestStore Proteus.keys.PreKey.generate_prekeys 0, (Proteus.session.Session.MAX_SESSION_STATES + 2)
+
+    bob_bundle = (i, store) -> Proteus.keys.PreKeyBundle.new bob_ident.public_key, store.prekeys[i]
+
+    alice = null
+    bob = null
+    hello_bob = null
+
+    Proteus.session.Session.init_from_prekey alice_ident, bob_bundle 1, bob_store
+    .then (s) ->
+      alice = s
+      alice.encrypt 'Hello Bob!'
+
+    .then (m) ->
+      hello_bob = m
+      assert_init_from_message bob_ident, bob_store, hello_bob, 'Hello Bob!'
+
+    .then (s) ->
+      bob = s
+      assert Object.keys(bob.session_states).length is 1
+
+      Promise.all([0..(Proteus.session.Session.MAX_SESSION_STATES - 1)].map((obj, i) ->
+        new Promise (resolve, reject) ->
+          Proteus.session.Session.init_from_prekey(alice_ident, bob_bundle(i + 2, bob_store))
+          .then (s) ->
+            alice = s
+            alice.encrypt 'Hello Bob!'
+
+          .then (m) ->
+            hello_bob = m
+            assert_decrypt 'Hello Bob!', bob.decrypt bob_store, hello_bob
+
+          .then((() -> resolve()), (err) -> reject(err))))
```

Comment:
Evaluate if this can be done with:

```coffeescript
.then resolve
.catch reject
```

Instead of having an army of brackets like `.then((() -> resolve()), (err) -> reject(err))))`.